File: src/ssr_pkg/src/yao_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char *argv[])
{
    ros::init(argc, argv, "yao_node");
    printf("this is yao_node~\n");

    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<std_msgs::String>("yao_topic", 10);

    ros::Rate loop_rate(10);

    while (ros::ok())
    {
        std_msgs::String msg;
        msg.data = "yao_node_publisher...";
        pub.publish(msg);
        loop_rate.sleep();
    }
    
    return 0;
}

File: src/ssr_pkg/scripts/yao_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String

if __name__ == "__main__":
    rospy.init_node("yao_node")
    rospy.logwarn("this is py_yao_node~")

    pub = rospy.Publisher("yao_topic", String, queue_size=10)

    loop_rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        msg = String()
        msg.data = "py_yao_node_publisher..."
        pub.publish(msg)

        loop_rate.sleep()

